xc7/utils/prjxray_form_channels.py

```diff
@@ -140,11 +144,11 @@ def get_switch_timing(
 
             write_cur.execute(
                 """
```

Comment:
You do need to update the `key` used above to include the penalty cost, or the switch cache may return the wrong value.

Reply:
Done